avm-transpiler/src/transpile_contract.rs

```diff
@@ -92,7 +92,7 @@ impl From<CompiledAcirContractArtifact> for TranspiledContractArtifact {
         let mut has_public_dispatch = false;
 
         for function in contract.functions {
-            if function.custom_attributes.contains(&"public".to_string()) {
+            if function.custom_attributes.contains(&"abi_public".to_string()) {
```

Comment:
Renamed the attribute from `public` to `abi_public` because now the dev never sees it and `abi_public` is used only to inform the artifact generation regarding what type of function it is.

`abi_public` is a much better name as it's way more searchable and it makes it clearer that now it's just a low-level thing.